src/app/rpmostree-ex-builtin-commit2jigdo.c

```diff
@@ -0,0 +1,1167 @@
+
+  guint64 oirpm_bytes_big = 0;
+  GLNX_HASH_TABLE_FOREACH_IT (new_reachable_big, it, const char *, checksum,
+                              void *, unused)
+    {
+      guint32 objsize;
+      if (!query_objsize_assert_32bit (self->repo, checksum, &objsize, error))
+        return FALSE;
+      g_assert_cmpint (objsize, >=, BIG_OBJ_SIZE);
+
+      g_autofree char *obj_contenthash =
+        contentonly_hash_for_object (self->repo, checksum, cancellable, error);
+      if (!obj_contenthash)
+        return FALSE;
+      g_autofree char *objsize_formatted = g_format_size (objsize);
+
+      /* This is complex to implement; it would be useful for the grub2-efi data
+       * but in the end we should avoid having content-identical objects in the
+       * OS data anyways.
+       */
+#if 0
+      const char *probable_source = g_hash_table_lookup (self->objsize_to_object, GUINT_TO_POINTER (objsize));
+
+      if (probable_source)
+        {
+          g_autofree char *probable_source_contenthash =
+            contentonly_hash_for_object (self->pkgcache_repo, probable_source, cancellable, error);
+          if (!probable_source_contenthash)
+            return FALSE;
+          if (g_str_equal (obj_contenthash, probable_source_contenthash))
+            {
+              g_print ("%s %s (pkg content hash hit)\n", checksum, objsize_formatted);
+              g_hash_table_replace (new_big_pkgidentical, g_strdup (checksum), g_strdup (probable_source));
+              g_hash_table_iter_remove (&it);
+
+              pkg_bytes += objsize;
+              /* We found a package content hash hit, loop to next */
+              continue;
+            }
+        }
+#endif
+
+      /* OK, see if it duplicates another *new* object */
+      GPtrArray *identicals = g_hash_table_lookup (new_big_content_identical, obj_contenthash);
+      if (!identicals)
+        {
+          identicals = g_ptr_array_new_with_free_func (g_free);
+          g_hash_table_insert (new_big_content_identical, g_strdup (obj_contenthash), identicals);
+          g_print ("%s %s (new, objhash %s)\n", checksum, objsize_formatted, obj_contenthash);
+          oirpm_bytes_big += objsize;
+        }
+      else
+        {
+          g_print ("%s (content identical with %u objects)\n", checksum, identicals->len);
+        }
+      g_ptr_array_add (identicals, g_strdup (checksum));
+    }
+
+  { g_autofree char *pkg_bytes_formatted = g_format_size (pkg_bytes);
+    g_autofree char *oirpm_bytes_formatted_small = g_format_size (oirpm_bytes_small);
+    g_autofree char *oirpm_bytes_formatted_big = g_format_size (oirpm_bytes_big);
+    g_print ("pkg content size: %s\n", pkg_bytes_formatted);
+    g_print ("oirpm content size (small objs): %s\n", oirpm_bytes_formatted_small);
+    g_print ("oirpm content size (big objs): %s\n", oirpm_bytes_formatted_big);
+  }
+
+  g_auto(GLnxTmpDir) oirpm_tmpd = { 0, };
```

Comment:
There's definitely a couple of spots where we could factor out clean self-contained functions. Though here at least seems like a good place to split this behemoth in two, WDYT? There's already a lot in scope to keep track of.

Reply:
Yep done